model/patch_lifecycle.go

```diff
@@ -349,6 +349,10 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		}).TVPairsToVariantTasks()
 	}
 
+	// if variant tasks is still empty, then the patch is empty and we shouldn't add to commit queue
+	if p.Alias == evergreen.CommitQueueAlias && len(p.VariantsTasks) == 0 {
+		return nil, errors.Errorf("No builds or tasks for commit queue version in projects '%s', githash '%s'", p.Project, p.Githash)
```

Comment:
is the thinking that when `FinalizePatch` returns an error the commit queue will dequeue the item?

Reply:
Yes, in both these cases we log an error (which seems reasonable, in case users are like "where did my patch go) and dequeue the item.